README.md

```diff
@@ -1,56 +1,114 @@
-# anchor-gpt
+# Anchor-gpt
 
-Diagnose and find hallucinations in your grounded Large Language Model prompts with Anchor-GPT!
+Find the worst hallucination prone prompts and use them to fine-tune / ground your LLM with Anchor-GPT
 
 ## Installation
 
 ```bash
 pip install anchor-gpt
 ```
 
-## Example Usage
+## Step by Step
+1. Use prompt logger to log your prompts and their grouding scores
+```python
+from anchor_gpt import PromptLogger, Prompt
+
+# Your regular grounding process
+prompt_embeddings = embedding_model.encode(prompt)
+index_response = my_index_endpoint.find_neighbors(
+    queries=prompt_embeddings,
+    num_neighbors=10,
+)
+
+grounding_data = []
+grounding_distances = []
+for grounding_index, grounding_distance in index_response:
+    grounding_data.append(my_index_endpoint.get(grounding_index))
+    grounding_distances.append(grounding_distance)
+
+grounded_prompt = build_prompt(prompt, grounding_data)
+
+# Call your LLM
+chat_response = my_llm.chat(grounded_prompt, temperature=0.1)
 
+
+# Log the prompt
+prompt_logger = PromptLogger()
+my_prompt = prompt_logger.log(Prompt(
+    text=prompt,
+    response=chat_response,
+    scores={'grounding_distances': grounding_distances},
+    embeddings=prompt_embeddings,
+))
+```
+
+2. Add additional scores like user feedback asynchronously
 ```python
+my_prompt.add_score({'user_feedback': 0.8})
```

Comment:
How do you add several user_feedbacks with this dict API? Does the user have to retrieve the existing array then push in the new score himself and pass that to `add_score`?

Reply:
the `add_score` api uses `update` now, so it overrides existing keys

Reply:
So no way to have several use feedbacks by the same name - ok

Reply:
Ok, say, several distances asynchonously